Sources/OutlineView/OutlineViewDataSource.swift

```diff
@@ -43,4 +89,151 @@ where Data.Element: Identifiable {
+    
+    func outlineView(
+        _ outlineView: NSOutlineView,
+        pasteboardWriterForItem item: Any
+    ) -> NSPasteboardWriting? {
+        
+        guard let writer = dragWriter,
+              let writeData = writer(typedItem(item).value)
+        else { return nil }
+        
+        return writeData
+    }
+    
+    private func dropTargetData(in outlineView: NSOutlineView, dragInfo: NSDraggingInfo, item: Any?, childIndex: Int) -> DropTarget<Data.Element>? {
+        guard let pasteboardItems = dragInfo.draggingPasteboard.pasteboardItems,
+              !pasteboardItems.isEmpty
+        else { return nil }
+        
+        let decodedItems = pasteboardItems.compactMap {
+            dropReceiver.readPasteboard(item: $0)
+        }
+        
+        let dropTarget = DropTarget<Data.Element>(
+            items: decodedItems,
+            intoElement: item.map(typedItem)?.value,
+            childIndex: childIndex == NSOutlineViewDropOnItemIndex ? nil : childIndex,
+            isItemExpanded: { item in
+                let typed = OutlineViewItem<Data>(value: item, children: self.childrenSource)
+                return outlineView.isItemExpanded(typed)
+            }
+        )
+        return dropTarget
+    }
+    
+    func outlineView(
+        _ outlineView: NSOutlineView,
+        validateDrop info: NSDraggingInfo,
+        proposedItem item: Any?,
+        proposedChildIndex index: Int
+    ) -> NSDragOperation {
+        
+        guard let dropTarget = dropTargetData(in: outlineView, dragInfo: info, item: item, childIndex: index)
+        else { return [] }
+        
+        let validationResult = dropReceiver.validateDrop(target: dropTarget)
+
+        switch validationResult {
+            
+        case .copy:
+            return .copy
+        case .move:
+            return .move
+        case .deny:
+            return []
+        case let .copyRedirect(item, childIndex):
+            let redirectTarget = item.map { OutlineViewItem<Data>(value: $0, children: childrenSource) }
+            outlineView.setDropItem(redirectTarget, dropChildIndex: childIndex ?? NSOutlineViewDropOnItemIndex)
+            return .copy
+        case let .moveRedirect(item, childIndex):
+            let redirectTarget = item.map { OutlineViewItem<Data>(value: $0, children: childrenSource) }
+            outlineView.setDropItem(redirectTarget, dropChildIndex: childIndex ?? NSOutlineViewDropOnItemIndex)
+            return .move
+        }
+        
+    }
+
+    func outlineView(
+        _ outlineView: NSOutlineView,
+        acceptDrop info: NSDraggingInfo,
+        item: Any?,
+        childIndex index: Int
+    ) -> Bool {
+        
+        guard let dropTarget = dropTargetData(in: outlineView, dragInfo: info, item: item, childIndex: index)
+        else { return false }
+        
+        // Perform `acceptDrop(target:)` to get result of drop
+        let res = dropReceiver.acceptDrop(target: dropTarget)
```

Comment:
It would be better have a more descriptive name for the variable here. One possible option is `dropSuccessful`.